File: week10/Medium.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Medium {

    public static int product(List<Integer> numbers) {
        int product = 1;

        for (int number : numbers) {
            product *= number;
        }

        return product;
    }

    public static void main(String[] args) {

        List<Integer> numbers = new ArrayList<>();

        try (Scanner sc = new Scanner(new File("input/medium.in"))) {

            while (sc.hasNextInt()) {
                numbers.add(sc.nextInt());
            }

        } catch (FileNotFoundException e) {
            System.err.println(e);
        }

        try (PrintWriter pw = new PrintWriter("output/medium.out")) {

            pw.println("Numbers: " + numbers);
            pw.println("Product: " + product(numbers));

        } catch (FileNotFoundException e) {
            System.err.println(e);
        }

    }

}

File: week10/medium.py
def product(numbers: list[int]) -> int:
    product = 1

    for number in numbers:
        product *= number
    
    return product


def main():
    numbers = []

    with open("input/medium.in", 'r') as f:
        for line in f:
            numbers.append(int(line))
    
    with open("output/medium.out", 'w') as f:
        f.write(f"Numbers: {numbers}\n")
        f.write(f"Product: {product(numbers)}\n")


if __name__ == '__main__':
    main()

